src/vector_index.c

```diff
@@ -293,35 +293,39 @@ static int VecSimIndex_validate_Rdb_parameters(RedisModuleIO *rdb, VecSimParams
     size_t old_block_size = 0;
     size_t old_initial_cap = 0;
     RedisModule_LogIOError(rdb, REDISMODULE_LOGLEVEL_WARNING, "ERROR: %s", QueryError_GetError(&status));
-    // We change the initial size to 0 and block size to default and try again.
-    // setting block size to 0 will later set it to default.
+    // We change the initial size and block size to default and try again.
     switch (vecsimParams->algo) {
       case VecSimAlgo_BF:
         old_block_size = vecsimParams->bfParams.blockSize;
         old_initial_cap = vecsimParams->bfParams.initialCapacity;
         vecsimParams->bfParams.blockSize = 0;
-        vecsimParams->bfParams.initialCapacity = 0;
+        vecsimParams->bfParams.initialCapacity = SIZE_MAX;
```

Comment:
why size_max and not 0?

Reply:
because 0 is a valid value for initial capacity, unlike block size which can get only >=1 value from the user